src/base/static/components/organisms/map-measurement-overlay.tsx

```diff
@@ -0,0 +1,438 @@
+  const positions = React.useRef<number[][]>([]);
+  const webMercatorViewport = React.useRef<WebMercatorViewport>();
+  const overlayRef = React.useRef<ExposedCanvasOverlay>(null);
+
+  const handleOverlayClick = evt => {
+    if (!selectedTool.current) {
+      return;
+    }
+
+    const { left, top } = evt.currentTarget.getBoundingClientRect();
+    // Get unprojected (i.e. lng/lat) coordinates from the x/y click position,
+    // relative to the overlay canvas element.
+    const unprojected =
+      webMercatorViewport.current &&
+      webMercatorViewport.current.unproject([evt.x - left, evt.y - top]);
+
+    unprojected && positions.current.push(unprojected);
+    selectedTool.current &&
+      setMeasurementFeatureCollection(
+        buildMeasurementFeatureCollection(
+          selectedTool.current,
+          positions.current,
+        ),
+      );
+  };
+
+  const handleUndoLastPoint = () => {
+    positions.current.pop();
+    selectedTool.current &&
+      setMeasurementFeatureCollection(
+        buildMeasurementFeatureCollection(
+          selectedTool.current,
+          positions.current,
+        ),
+      );
+  };
+
+  const handleReset = () => {
+    positions.current = [];
+    setMeasurementFeatureCollection(featureCollection([]));
+    selectedTool.current = null;
+    setUnits(null);
+    setMeasurement(null);
+    if (overlayRef.current) {
+      overlayRef.current._canvas.style.pointerEvents = "none";
+    }
+  };
+
+  React.useEffect(() => {
+    // Unless I'm missing the right way to use overlays, it seems necessary
+    // to attach a click listener here to capture clicks on the overlay.
+    // See: https://github.com/uber/react-map-gl/issues/470
+    if (overlayRef && overlayRef.current && overlayRef.current._canvas) {
+      overlayRef.current._canvas.addEventListener("click", handleOverlayClick);
+      // This direct manipulation of the `pointer-events` style property feels
+      // like a hack, but the goal is to enable clicks on the underlying map
+      // when no measurement tool is selected.
+      overlayRef.current._canvas.style.pointerEvents = "none";
+    } else {
+      // eslint-disable-next-line no-console
+      console.error(
+        "Measurement overlay: failed to attach click event listener",
+      );
+    }
+
+    return () => {
+      overlayRef &&
+        overlayRef.current &&
+        overlayRef.current._canvas.removeEventListener(
+          "click",
+          handleOverlayClick,
+        );
+    };
+  }, []);
+
+  React.useEffect(
+    () => {
+      webMercatorViewport.current = getNewViewport({
+        ...props.viewport,
+        width: props.width,
+        height: props.height,
+      });
+    },
+    [props.viewport, props.width, props.height],
+  );
+
+  React.useEffect(
+    () => {
+      const {
+        features,
+      }: { features: Feature[] } = measurementFeatureCollection;
+      const measurementFeature = features[features.length - 1] as Feature;
```

Comment:
Nit: I don't think this type cast is needed:

```suggestion
      const measurementFeature = features[features.length - 1];
```